### PR TITLE
#165124861 Add User Location Automatically upon User Registration

### PR DESCRIPTION
**Pull request by verenceLola**
### What does this PR do?
Add user location automatically upon user registration 
### Description of the task to be completed?
Upon successful user registration, update the location table to add the location of the user if it doesn't exist in the database.
 - Clone the branch and follow the following steps to [setup](https://github.com/andela/mrm_api).
 - Checkout by running `git checkout ft-add-location-name-165124861`
 - Perform any query or mutation with a valid token from a new user(not already in the database)
 - Check your users table to make sure that the user was successfully added. Now open the location table to confirm that the new user location has also been added.

### Any background context you want to provide?
Currently, when a new user is added into the database, the location of the user has to be manually added to the location table. To prevent this, upon user registration, the location of the user should be added automatically into the database.
Since the user information only has the location name and not other fields present in the location table, constraints such as NOT NULL should be removed to permit NULL values in fields such as `abbreviation`. 
### What are the relevant pivotal tracker stories?
[#165124861](https://www.pivotaltracker.com/story/show/165124861)

### CODE REVIEW

File: api/location/models.py [hunk 1]
```diff
@@ -23,15 +23,14 @@ class Location(Base, Utility):
     id = Column(Integer, Sequence('locations_id_seq',
                                   start=1, increment=1), primary_key=True)
     name = Column(String, nullable=False)
-    abbreviation = Column(String, nullable=False)
```
Comment (RachaelNantale) [inline api/location/models.py]: Great work @verenceLola, why did you remove this?
same as line 37

Comment (verenceLola) [reply api/location/models.py]: Kindly have a look at the `background context` of the PR description. I've updated the information.

File: helpers/auth/authentication.py [hunk 2]
```diff
@@ -135,9 +136,15 @@ def wrapper(*args, **kwargs):
                             if value['location']:
                                 user.location = value['location']['name']
                                 user.save()
+                                add_location(
+                                    value['location']['name']
+                                )  # add location name if not in database
                             else:
                                 user.location = "Nairobi"
```
Comment (KMaina) [inline helpers/auth/authentication.py]: Good work, why are we defaulting to Nairobi here?

File: helpers/auth/authentication.py [hunk 3]
```diff
@@ -129,15 +125,25 @@ def wrapper(*args, **kwargs):
...
                                 user.save()
+                                add_location(
+                                    "Nairobi"
```
Comment (TeamoreA) [inline helpers/auth/authentication.py]: Good work @verenceLola . However, I didn't get the reason behind defaulting the location to Nairobi yet your PR should aim at saving the user's exact location.

Comment (verenceLola) [reply helpers/auth/authentication.py]: Still, if the location of the user cannot be fetched from  the Andela API, it's defaulted to `Nairobi` as before. @TeamoreA

Comment (TeamoreA) [reply helpers/auth/authentication.py]: Alright👍🏼

Comment (joshuaocero) [reply helpers/auth/authentication.py]: @verenceLola take this opportunity to refactor the duplication in the `if..else` statement. Remove the `else`. Can we also make the `add_location` function name more descriptive? Because it does not only add the location, it also performs a check. Otherwise well done!

**Comment by verenceLola** [comment]
> Kindly check the structure of your commit message.

Thanks for the feedback @Bonifase .This has been rectified
